src/Sylius/Bundle/UserBundle/Controller/UserController.php

```diff
@@ -105,7 +110,91 @@ public function resetPasswordAction(Request $request, $token)
         );
     }
 
-    protected function prepareResetPasswordRequest(Request $request, TokenProviderInterface $generator, $senderEvent)
+    /**
+     * @param Request $request
+     * @param string $token
+     *
+     * @return Response
+     */
+    public function verifyAction(Request $request, $token)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+        $redirectRoute = $request->attributes->get('_sylius[redirect]', 'sylius_shop_homepage', true);
```

Comment:
``sylius_shop_homepage`` should not be in UserBundle. It should be configured as redirect in shop routing cfgs.